### PR TITLE
Add few System.Reflection APIs

### PR DESCRIPTION
**Pull request by sepidehMS**
Adding remaining exposable System.Reflection APIs. 
Related issue: https://github.com/dotnet/corefx/issues/11320

### CODE REVIEW

**Comment by sepidehMS** [comment]
@dotnet-bot test OSX x64 Checked Build and Test please

**Comment by jkotas** [comment]
- AssemblyNameProxy should be implemented in corefx instead so that it can be shared with CoreRT
- None of the methods that taked TypedReference argument should be added. They are out of .NET Standard 2.0 (cc @weshaggard)

**Comment by jkotas** [comment]
Also, all IsSecurity* methods in reflection namespace should be changed to returned hardcoded values in CoreCLR: `IsSecurityCritical => true; IsSecuritySafeCritical =>false; IsSecurityTransparent => false`.

**Comment by sepidehMS** [comment]
@jkotas About AssemblyNameProxy, sorry I forgot about that. I added it to an issue.
The methods with TypedReference argument are listed in the netstandard2.0 surface area [here](https://raw.githubusercontent.com/dotnet/standard/master/netstandard/ref/mscorlib.cs?token=ANn1B6UpsSnK0AXF44ymGhrfJsDkxXPQks5X3DeiwA%3D%3D). Should I still not include them?
I'll change the IsSecurity* APIs.

**Comment by sepidehMS** [comment]
@dotnet-bot test Linux ARM Emulator Cross Debug Build please

**Comment by weshaggard** [comment]
@sepidehMS I recently cleaned out those from netstandard (https://github.com/dotnet/standard/commit/6e3ad137e2b4e6556ca9467b5e41e887554e944c).

**Comment by sepidehMS** [comment]
@dotnet-bot test Windows_NT arm Cross Checked Build please
@dotnet-bot test Windows_NT arm Cross Debug Build please
@dotnet-bot test Windows_NT arm Cross Release Build please

**Comment by sepidehMS** [comment]
@mmitche Are these failures expected?

**Comment by mmitche** [comment]
@sepidehMS It looks like the non-PR job is currently failing with the same error.
@mikem8361 @gkhanna79 Do these look familiar?

http://dotnet-ci.cloudapp.net/job/dotnet_coreclr/job/master/job/arm_cross_checked_windows_nt/2/

**Comment by gkhanna79** [comment]
@mmitche Does this machine was Windows SDK installed?

**Comment by mmitche** [comment]
@gkhanna79 I would imagine so.  Wouldn't that be part of the standard VS C++ install?  Not sure you can hop on and check.

**Comment by gkhanna79** [comment]
I am reverting the CI change for now (https://github.com/dotnet/coreclr/pull/7213) and will hope in later to confirm what is going on.

**Comment by gkhanna79** [comment]
@mmitche Can you point me to the machine as well?

**Comment by gkhanna79** [comment]
Ihave reverted the change - please try the PR again.

**Comment by sepidehMS** [comment]
Thanks @gkhanna79

**Comment by sepidehMS** [comment]
@dotnet-bot test Windows_NT arm Cross Checked Build please
@dotnet-bot test Windows_NT arm Cross Debug Build please
@dotnet-bot test Windows_NT arm Cross Release Build please

**Comment by gkhanna79** [comment]
No worries - Apologies for the inconvenience @sepidehMS

**Comment by sepidehMS** [comment]
@gkhanna79 Do I need to create a different PR so the Windows_NT Arm builds are not triggered?

File: src/mscorlib/src/System/Reflection/ConstructorInfo.cs [hunk 1]
```diff
@@ -664,17 +664,29 @@ public override MethodBody GetMethodBody()
 
         public override bool IsSecurityCritical
         {
+#if FEATURE_CORECLR
+            get { return true; }
+#else
             get { return RuntimeMethodHandle.IsSecurityCritical(this); }
+#endif
```
Comment (sepidehMS) [inline src/mscorlib/src/System/Reflection/ConstructorInfo.cs]: @jkotas Did I get what you meant correctly?

File: src/vm/ecalllist.h [hunk 2]
```diff
@@ -548,9 +548,9 @@ FCFuncStart(gRuntimeMethodHandle)
     FCFuncElement("GetMethodBody", RuntimeMethodHandle::GetMethodBody)    
 #ifndef FEATURE_CORECLR
     FCFuncElement("_IsTokenSecurityTransparent", RuntimeMethodHandle::IsTokenSecurityTransparent)
+#endif // FEATURE_CORECLR
```
Comment (jkotas) [inline src/vm/ecalllist.h]: This change should be undone - the managed side is not calling the FCalls anymore

File: src/mscorlib/src/System/Reflection/ConstructorInfo.cs (hunk 1 above)
Comment (jkotas) [inline src/mscorlib/src/System/Reflection/ConstructorInfo.cs]: Yes!

File: src/vm/ecalllist.h (hunk 2 above)
Comment (sepidehMS) [inline src/vm/ecalllist.h]: Done

**Pull request merged**